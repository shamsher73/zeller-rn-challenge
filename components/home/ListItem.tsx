import { StyleSheet, View } from 'react-native';
import LetterBox from '../LetterBox';
import { ThemedText } from '../ThemedText';
import { capitalizeWords } from '@/utils';
import { User } from '@/types';

export default function ListItem({ user }: { user: User }) {
    return (
        <View style={styles.container}>
            <LetterBox character={user.name.charAt(0)} />
            <View style={styles.subContainer}>
                <ThemedText type='primary'>{user.name}</ThemedText>
                <ThemedText type='secondary'>{capitalizeWords(user.role)}</ThemedText>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        marginVertical: 10
    },
    subContainer: {
        marginLeft: 10,
        justifyContent: 'space-between'
    },
});
