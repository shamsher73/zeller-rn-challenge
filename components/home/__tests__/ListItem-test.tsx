import ListItem from '@/components/home/ListItem';
import { render } from '@testing-library/react-native';

describe('<ListItem />', () => {

    it('should render correctly when provided with valid user data', () => {
        const user = { id: "123", name: "David Miller", role: "ADMIN" }
        const { getByText } = render(<ListItem user={user} />);

        expect(getByText('David Miller')).toBeTruthy();
        expect(getByText('Admin')).toBeTruthy();
    });

    it('should handle user with an empty name string', () => {
        const user = { id: "123", name: '', role: 'ADMIN' };
        const { getByText } = render(<ListItem user={user} />);

        expect(getByText('Admin')).toBeTruthy();
    });

    it('should display first character of user name in LetterBox', () => {
        const user = { id: "123", name: "David Miller", role: "ADMIN" }
        const { getByText } = render(<ListItem user={user} />);

        expect(getByText('D')).toBeTruthy();
    });

    it('should render correctly when provided with user with an empty role string', () => {
        const user = { id: "123", name: 'David Miller', role: '' };
        const { getByText } = render(<ListItem user={user} />);

        expect(getByText('David Miller')).toBeTruthy();
        expect(getByText('')).toBeTruthy();
    });

    test('snapshot', () => {
        const user = { id: "123", name: "David Miller", role: "ADMIN" }
        const tree = render(<ListItem user={user} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
