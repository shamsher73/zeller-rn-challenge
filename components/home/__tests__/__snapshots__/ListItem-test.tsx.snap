// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e8f2fb",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0370ce",
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": 800,
        }
      }
    >
      D
    </Text>
  </View>
  <View
    style={
      {
        "justifyContent": "space-between",
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        [
          undefined,
          undefined,
          {
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
          undefined,
        ]
      }
    >
      David Miller
    </Text>
    <Text
      style={
        [
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 12,
            "fontWeight": "300",
          },
          undefined,
        ]
      }
    >
      Admin
    </Text>
  </View>
</View>
`;
